### PR TITLE
Private template support in the SdkGenerator automation scripts

### PR DESCRIPTION
**Pull request by pgilmorepf**

### CODE REVIEW

File: JenkinsConsoleUtility/JenkinsScripts/testInit.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@ ResetRepo (){
     git checkout master || git checkout -b master || CleanCurrentRepo
     git pull origin master
 
+    # Delete $GitDestBranch, reset it to master, prep for next build and fresh write
     if [ "$GitDestBranch"!="master" ]; then
         git fetch --progress origin
         git branch -D $GitDestBranch || true
```
Comment (Vicodex) [inline JenkinsConsoleUtility/JenkinsScripts/testInit.sh]: > || true [](start = 36, length = 8)

what is this for? "II true"

**Comment by Vicodex** [comment]
>         git checkout $GitDestBranch

Is this line necessary? you are already on that branch

---
Refers to: JenkinsConsoleUtility/JenkinsScripts/testInit.sh:35 in 31d5caa. [](commit_id = 31d5caa3b35ca22c74c88fe5d95699703d389014, deletion_comment = False)

File: JenkinsConsoleUtility/JenkinsScripts/testInit.sh [hunk 2]
```diff
@@ -27,6 +27,7 @@ ResetRepo (){
     git checkout master || git checkout -b master || CleanCurrentRepo
     git pull origin master
 
+    # Delete $GitDestBranch, reset it to master, prep for next build and fresh write
     if [ "$GitDestBranch"!="master" ]; then
```
Comment (Vicodex) [inline JenkinsConsoleUtility/JenkinsScripts/testInit.sh]: >"$GitDestBranch"!="master" [](start = 9, length = 26)

Why are we not fetching the latest master? what if it was updated in github?

File: JenkinsConsoleUtility/JenkinsScripts/testInit.sh [hunk 3]
```diff
@@ -27,6 +27,7 @@ ResetRepo (){
...
         git fetch --progress origin
```
Comment (Vicodex) [inline JenkinsConsoleUtility/JenkinsScripts/testInit.sh]: >git fetch --progress origin [](start = 8, length = 27)

This line doesn't seem to be necessary, we just fetched all the latest stuff above with the same command

File: JenkinsConsoleUtility/JenkinsScripts/testInit.sh [hunk 4]
```diff
@@ -40,10 +41,16 @@ DoWork () {
...
+    # We sync master on the SHARED_WORKSPACE
+    SyncGitHubRepo "$SHARED_WORKSPACE" "SDKGenerator" "SDKGenerator"
     SyncGitHubRepo "$SHARED_WORKSPACE/sdks" "$SdkName"
+    # We sync $GitSdkGenBranch in the real workspace
     SyncWorkspaceRepo "$SHARED_WORKSPACE" "$WORKSPACE" "SDKGenerator" "SDKGenerator" "$GitSdkGenBranch"
-    SyncWorkspaceRepo "$SHARED_WORKSPACE/sdks" "$WORKSPACE/sdks" "$SdkName"
+    SyncWorkspaceRepo "$SHARED_WORKSPACE/sdks" "$WORKSPACE/sdks" "$SdkName" "$GitSdkDestBranch"
+
+    if [ ! -z "$SdkGenPrvTmplRepo" ]; then
+        SyncGitHubRepo "$WORKSPACE/SDKGenerator/privateTemplates" "$SdkGenPrvTmplRepo" "$GitPrvTmplBranch"
```
Comment (Vicodex) [inline JenkinsConsoleUtility/JenkinsScripts/testInit.sh]: >$SdkGenPrvTmplRepo [](start = 67, length = 18)

We probably should have called it $GitPrvTmplRepo for consistency

File: SDKBuildScripts/shared_build.sh [hunk 5]
```diff
@@ -36,7 +36,9 @@ CleanCodeFiles () {
 BuildSdk () {
     pushd ..
     echo === SHARED BUILDING $SdkName ===
-    if [ -z "$targetSrc" ]; then
+    if [ ! -z "$SdkGenPrvTmplRepo" ]; then
+        node generate.js $SdkGenPrvTmplRepo=$destPath $apiSpecSource $SdkGenArgs $buildIdentifier $VerticalNameInternal
```
Comment (Vicodex) [inline SDKBuildScripts/shared_build.sh]: >$SdkGenPrvTmplRepo [](start = 25, length = 18)

$destPath is where the generated SDK will be placed, right?

$SdkGenPrvTmplRepo sounds like "private template repo", i.e. a repo where the template resides.. I'm a bit confused.

File: JenkinsConsoleUtility/JenkinsScripts/testInit.sh (hunk 1 above)
Comment (pgilmorepf) [reply JenkinsConsoleUtility/JenkinsScripts/testInit.sh]: I am telling Jenkins that if this line reports an error-code, that's ok, don't fail the Job.

File: SDKBuildScripts/shared_build.sh (hunk 5 above)
Comment (pgilmorepf) [reply SDKBuildScripts/shared_build.sh]: The "=" does not mean assignment, it's just clumsy legacy syntax.
Honestly, the discussion is lengthy, and will soon be redundant when we get genConfig.json online.
For now: It's not assignment, despite the appearance that it is. It's just crappy syntax that we'll remove soon.

**Pull request merged**